src/full/Agda/TypeChecking/Serialise/Instances/Internal.hs

```diff
@@ -232,11 +242,22 @@ instance EmbPrj NotBlocked where
     valu [3, a] = valuN MissingClauses a
     valu _      = malformed
 
-instance EmbPrj Blocked_ where
-  icod_ (NotBlocked a b) = icodeN' NotBlocked a b
-  icod_ Blocked{} = __IMPOSSIBLE__
+-- Andreas, 2024-01-02, issue #7044.
+-- We only serialize 'neverUnblock';
+-- other than that, there should not be any blockers left at serialization time.
+blockedToMaybe :: Blocked_ -> Maybe NotBlocked
+blockedToMaybe = \case
+  NotBlocked a ()       -> Just a
+  Blocked a ()
+    | a == neverUnblock -> Nothing
+    | otherwise         -> __IMPOSSIBLE__
 
-  value = valueN NotBlocked
+blockedFromMaybe :: Maybe NotBlocked -> Blocked_
+blockedFromMaybe = maybe (Blocked neverUnblock ()) (`NotBlocked` ())
+
+instance EmbPrj Blocked_ where
+  icod_ = icod_ . blockedToMaybe
+  value = blockedFromMaybe <.> value
```

Comment:
@AndrasKovacs : Could I have serialized `Blocked_` efficiently without going through `Maybe`?  
I didn't see how to do this with the `icodeN'` interface.  I want something like
```haskell
  icod_ (NotBlocked a ()) = icodeN' NotBlocked a
  icod_ (Blocked a ()) | a == neverUnblock = icodeN' Blocked

  valu [a] = valuN NotBlocked a ()
  valu [] = valuN Blocked neverUnblock ()
````
(but this is of course not type correct).

Reply:
I also don't see a better way with `icodeN` and `valuN`. Doesn't really matter for efficiency IMO. I'll refactor the whole serializer anyway in the near future.